### PR TITLE
[Foundation] Add missing bindings in NSTask

### LINKED ISSUE #14659: NSTask launchAndReturnError not bound
I am working with NSTask and am currently using [launch](https://developer.apple.com/documentation/foundation/nstask/1414189-launch) to launch a NSTask. However, that method is deprecated, and [launchAndReturnError](https://developer.apple.com/documentation/foundation/nstask/2890105-launchandreturnerror) is recommended. That method has not been bound yet, however

```
Visual Studio Community 2022 for Mac Preview
Version 17.0 Preview (17.0 build 8320)
Installation UUID: 4176a999-a96a-437d-ad9b-ae89ecbde669

Runtime
.NET 6.0.3 (64-bit)
Architecture: Arm64

Xamarin Designer
Version: 17.1.5.47
Hash: 626674a78
Branch: remotes/origin/d17-0-vsmac-p8-candidate
Build date: 2022-03-30 15:01:38 UTC

Roslyn (Language Service)
4.1.0-3.22075.3+592501cbb9c9394072a245c15b3458ff88155d85

NuGet
Version: 6.0.0.262

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/6.0.201/Sdks
SDK Versions:
	6.0.201
	6.0.100-rc.1.21463.6
	5.0.402 (x64 – Unsupported)
	5.0.401 (x64 – Unsupported)
	5.0.400 (x64 – Unsupported)
	5.0.302 (x64 – Unsupported)
	5.0.301 (x64 – Unsupported)
	5.0.203 (x64 – Unsupported)
	5.0.202 (x64 – Unsupported)
	5.0.201 (x64 – Unsupported)
	3.1.414 (x64 – Unsupported)
	3.1.413 (x64 – Unsupported)
	3.1.412 (x64 – Unsupported)
	3.1.411 (x64 – Unsupported)
	3.1.410 (x64 – Unsupported)
	3.1.409 (x64 – Unsupported)
	3.1.408 (x64 – Unsupported)
	3.1.407 (x64 – Unsupported)
MSBuild SDKs: /usr/local/share/dotnet/sdk/6.0.201/Sdks

.NET SDK (x64)
SDK Versions:
	6.0.201
	6.0.200
	5.0.406
	5.0.405
	3.1.417
	3.1.416

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.3
	6.0.0-rc.1.21451.13
	5.0.11 (x64 – Unsupported)
	5.0.10 (x64 – Unsupported)
	5.0.9 (x64 – Unsupported)
	5.0.8 (x64 – Unsupported)
	5.0.7 (x64 – Unsupported)
	5.0.6 (x64 – Unsupported)
	5.0.5 (x64 – Unsupported)
	5.0.4 (x64 – Unsupported)
	3.1.20 (x64 – Unsupported)
	3.1.19 (x64 – Unsupported)
	3.1.18 (x64 – Unsupported)
	3.1.17 (x64 – Unsupported)
	3.1.16 (x64 – Unsupported)
	3.1.15 (x64 – Unsupported)
	3.1.14 (x64 – Unsupported)
	3.1.13 (x64 – Unsupported)

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	6.0.3
	6.0.2
	5.0.15
	5.0.14
	3.1.23
	3.1.22

Xamarin.Profiler
'/Applications/Xamarin Profiler.app' not found

Updater
Version: 11

Apple Developer Tools
Xcode 13.3 (20102)
Build 13E113

Xamarin.Mac
Version: 8.8.0.0 (Visual Studio Community)
Hash: f10d9e023
Branch: xcode13.3
Build date: 2022-03-15 11:47:34-0400

Xamarin.iOS
Version: 15.8.0.0 (Visual Studio Community)
Hash: f10d9e023
Branch: xcode13.3
Build date: 2022-03-15 11:47:35-0400

Xamarin.Android
Not Installed

Microsoft Build of OpenJDK
Java SDK: Not Found

Eclipse Temurin JDK
Java SDK: Not Found

Android SDK Manager
Version: 17.2.0.30
Hash: 1f76363
Branch: remotes/origin/d17-2
Build date: 2022-03-30 15:01:43 UTC

Android Device Manager
Version: 0.0.0.1109
Hash: cb5ab08
Branch: remotes/origin/d17-2
Build date: 2022-03-30 15:01:43 UTC

Build Information
Release ID: 1700008320
Git revision: 06baf68a7e5a3fd9ab6e343e85745d6f118f697a
Build date: 2022-03-30 14:58:01+00
Build branch: release-17.0
Build lane: release-17.0

Operating System
Mac OS X 12.3.1
Darwin 21.4.0 Darwin Kernel Version 21.4.0
    Fri Mar 18 00:47:26 PDT 2022
    root:xnu-8020.101.4~15/RELEASE_ARM64_T8101 arm64
```

### PR DESCRIPTION
**Pull request by haritha-mohan**
Added new bindings for deprecated methods/properties: 

Launch -> LaunchAndReturnError
LaunchFromPath -> LaunchFromUrl
LaunchPath -> ExecutableUrl
CurrentDirectoryPath -> CurrentDirectoryUrl

Fixes #14659

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]
Looks good! But I think there's a few more NSTask APIs we might want to implement:

* terminationHandler: https://developer.apple.com/documentation/foundation/nstask/1408746-terminationhandler
* qualityOfService: https://developer.apple.com/documentation/foundation/nstask/1415794-qualityofservice

Whenever we're implementing missing APIs for a class, it's often a good idea to check our xtro tests to see if we fixed everything we know about:

```
$ rolf@ ~/work/maccore/main/xamarin-macios/tests/xtro-sharpie $ gg NSTask
api-annotations-dotnet/MacCatalyst-Foundation.ignore:!missing-selector! NSTask::qualityOfService not bound
api-annotations-dotnet/MacCatalyst-Foundation.ignore:!missing-selector! NSTask::setQualityOfService: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! +NSTask::launchedTaskWithExecutableURL:arguments:error:terminationHandler: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::currentDirectoryURL not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::executableURL not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::launchAndReturnError: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::qualityOfService not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::setCurrentDirectoryURL: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::setExecutableURL: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::setQualityOfService: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::setTerminationHandler: not bound
api-annotations-dotnet/macOS-Foundation.ignore:!missing-selector! NSTask::terminationHandler not bound
macOS-Foundation.ignore:!missing-selector! +NSTask::launchedTaskWithExecutableURL:arguments:error:terminationHandler: not bound
macOS-Foundation.ignore:!missing-selector! NSTask::currentDirectoryURL not bound
macOS-Foundation.ignore:!missing-selector! NSTask::executableURL not bound
macOS-Foundation.ignore:!missing-selector! NSTask::launchAndReturnError: not bound
macOS-Foundation.ignore:!missing-selector! NSTask::qualityOfService not bound
macOS-Foundation.ignore:!missing-selector! NSTask::setCurrentDirectoryURL: not bound
macOS-Foundation.ignore:!missing-selector! NSTask::setExecutableURL: not bound
macOS-Foundation.ignore:!missing-selector! NSTask::setQualityOfService: not bound
macOS-Foundation.ignore:!missing-selector! NSTask::setTerminationHandler: not bound
macOS-Foundation.ignore:!missing-selector! NSTask::terminationHandler not bound
```

File: src/foundation.cs [hunk 1]
```diff
@@ -14866,6 +14866,9 @@ interface NSTask {
 		[Export ("launch")]
 		void Launch ();
 
+		[Export ("launchAndReturnError:")]
+		bool LaunchAndReturnError ([NullAllowed] out NSError error);
```
Comment (rolfbjarne) [inline src/foundation.cs:14870]: C# has method overloading, so this could be just:

```suggestion
		bool Launch ([NullAllowed] out NSError error);
```

**Review by tj-devel709 — APPROVED** [review]
Is there any todo files or ignores that these bindings you added can be removed from?

**Comment by haritha-mohan** [comment]
> Is there any todo files or ignores that these bindings you added can be removed from?

yep, they are removed now!

**Review by rolfbjarne — APPROVED** [review]

**Comment by haritha-mohan** [comment]
Test failure unrelated (Rolf is currently working on a fix, Mac Catalyst apps not launching properly)

**Pull request merged**